### ISSUE TITLE
Build Assertion 4: LIT Holder

### ISSUE THREAD

**Issue by StamfordDigital** [litentry/litentry-parachain#1030]
Check whether any of the linked accounts holds a minimum amount (0.000001) of LIT NOW. 

If TRUE, a VC should be generated and returned to the F/E. ?

----------------------------------------------------------------------------------------
Old description obsolate：

Build **Assertion 4: LIT Holder** which will generate a **Verifiable Credential (VC)** if the following validates:
- Check the IDHub controller (user) holds LIT tokens
  - **Important:** A specific address must be provided and passed to the Parachain from the F/E using the `params` extrinsic param to verify this
- If the ruleset validates then a VC should be generated and returned to the F/E.

**Comment by jingleizhang** [comment]
Correct: this ruleset should test if “any of the linked accounts holds any LIT”. Not just the main address.

**Comment by Kailai-Wang** [comment]
This makes less sense though, as the main address has to hold some LIT to be able to send extrinsics at all. That's why I consider this rule a bit vague. 

I suggest that we double-check it with Kayla or Rodrigo

**Comment by jingleizhang** [comment]
The actual workflow is:

- user calls the parachain extrinsic
- parachain&TEE knows this user’s main address (Litentry parachain address)
- because any active Litentry address needs to hold more than zero LIT tokens
- TEE queries user’s any other linked and verified address(e.g, ETH address, BSC address)
- and calculate ‘if they hold any LIT’, if true, TEE will generate a LIT Holder VC

**Comment by jingleizhang** [comment]
Reply from Rodrigo: 

```
The end result is we want to have a VC that says the user is a LIT Holder. Originally I was told that it would be too “Expensive” or too much work to check all accounts, so a specific account would be better to check. But if all accounts is easier/better, lets do that!

For the user the Quest would be:

- Connect any account that holds LIT (if you don’t have any already connected)
- Generate a VC to show that you are a LIT Holder

so in the VC generation we could check all
```

**Comment by Kailai-Wang** [comment]
> so in the VC generation we could check all

So it actually means we want to check if **any** linked account (so main account excluded) holds **any** LIT

**Comment by jingleizhang** [comment]
Tim gave the demo interface:

[https://graph.tdf-labs.io/latest/graphql?query=query{VerifiedCredentialsIsHodler(%0A%[…]%22%0A%20%20minimumBalance%3A%200.00000056%0A){isHodler}}%0A](https://graph.tdf-labs.io/latest/graphql?query=query{VerifiedCredentialsIsHodler(%0A%20%20addresses%3A%20[%220x61f2270153bb68dc0ddb3bc4e4c1bd7522e918ad%22]%2C%20%0A%20%20fromDate%3A%222022-10-16T00%3A00%3A00Z%22%2C%0A%20%20network%3A%20ethereum%2C%0A%20%20tokenAddress%3A%220xb59490aB09A0f526Cc7305822aC65f2Ab12f9723%22%0A%20%20minimumBalance%3A%200.00000056%0A){isHodler}}%0A)

[https://graph.tdf-labs.io/latest/graphql?query=query%7BVerifiedCredentialsIsHodler(%0[…]%20%20minimumBalance%3A%200.00000057%0A)%7BisHodler%7D%7D%0A](https://graph.tdf-labs.io/latest/graphql?query=query%7BVerifiedCredentialsIsHodler(%0A%20%20addresses%3A%20%5B%220x61f2270153bb68dc0ddb3bc4e4c1bd7522e918ad%22%5D%2C%20%0A%20%20fromDate%3A%222022-10-16T00%3A00%3A00Z%22%2C%0A%20%20network%3A%20ethereum%2C%0A%20%20tokenAddress%3A%220xb59490aB09A0f526Cc7305822aC65f2Ab12f9723%22%0A%20%20minimumBalance%3A%200.00000057%0A)%7BisHodler%7D%7D%0A)

**Comment by BillyWooo** [comment]
Discussed with @kaylawangnow , this requirement will be merged with #1024 . Both are querying LIT. Only give different timestamp as parameter.
